core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -17,27 +17,62 @@
  * functions on $window to be mocked in unit tests.
  */
 
-oppia.factory('UrlService', ['$window', function($window) {
-  return {
-    getUrlParams: function() {
-      var params = {};
-      var parts = $window.location.href.replace(
-        /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
-          params[key] = value;
+oppia.factory('UrlService', [
+  '$window', '$location',
+  function(
+      $window, $location) {
+    return {
+      // This function is for testing purposes (to mock $window.location)
+      getCurrentUrl: function() {
+        return $window.location;
+      },
+      getUrlParams: function() {
+        var params = {};
+        var parts = this.getCurrentUrl().href.replace(
+          /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
+            params[key] = value;
+          }
+        );
+        return params;
+      },
+      isIframed: function() {
+        var pathname = this.getPathname();
+        var urlParts = pathname.split('/');
+        return urlParts[1] === 'embed';
+      },
+      getPathname: function() {
+        return this.getCurrentUrl().pathname;
+      },
+      getParentExplorationIds: function() {
+        if ($location.search().parent) {
+          if ($location.search().parent.constructor === Array) {
+            return $location.search().parent;
+          } else if ($location.search().parent) {
+          /* By default, single parent id gave a single string whereas, an array
+             with  a single element is what is required, hence the additional
+             condition */
+            return [$location.search().parent];
+          }
+        } else {
+          return null;
         }
-      );
-      return params;
-    },
-    isIframed: function() {
-      var pathname = this.getPathname();
-      var urlParts = pathname.split('/');
-      return urlParts[1] === 'embed';
-    },
-    getPathname: function() {
-      return $window.location.pathname;
-    },
-    getHash: function() {
-      return $window.location.hash;
-    }
-  };
-}]);
+      },
+      /* parameterList is an array of exploration ids from which 1 is popped out
+         and then, URL updated. */
+      updateParameterList: function(parameterList) {
+        var parameterString = '#?parent=';
```

Comment:
I don't understand why we're using the "#"? Shouldn't it just be a regular query parameter (i.e. only using ? and &)?

Reply:
I am not sure but $location.search() returned correct value only when # was there, else it just returned null or undefined

Reply:
Then maybe you shouldn't be using location.search. There is probably some other function that retrieves query params (rather than hash params).

Reply:
Also if angular is doing the wrong thing, then I think a jquery solution is OK here. But I suggest seeing if it's possible to fix the angular version first. Hashes in URLs mean different things from query params -- the former are client-side and not meant for the server, which is not going to work here.

Reply:
I just searched, and this is an actual issue for $location, https://github.com/angular/angular.js/issues/7239 , there is a workaround enabling HTML5 for $locationProvider, should I try that?

Reply:
Done, not using location.search() anymore.